compare/image.c

```diff
@@ -79,8 +92,6 @@ bool read_png(const char *path, Image16 *img)
     ptrdiff_t half = 4 * w;
     if (depth == 8)
         ptr += half;
-    else
-        png_set_swap(png);
```

Comment:
I think that using an already existing infrastructure (`png_set_swap`) is better too. Also maybe autotools already have some standard tool for detecting endianness? It could be useful for big-endian ARM assembly too.

Reply:
CPUs with runtime-modifiable endianness exist. Why limit ourselves when it’s so easy to make this work universally?

Reply:
Code still has to be compiled for a particular endianness on those systems; your `read_16be` will lower to either a straight load or a bswap depending on whether it's built for big- or little-endian, even on e.g. ARM (where endianness is indeed runtime-configurable).

Reply:
In that case I think it's better to use `png_set_swap` anyway:
```c
static inline bool is_little_endian()
{
    uint16_t dst;
    uint8_t src[2] = { 1, 0 };
    memcpy(&dst, src, sizeof(dst));
    return dst == 1;

    // or use htons(), but it's additional dependency with different headers under *nix/windows
    // and looks like MSVC cannot inline it
}

...

if (is_little_endian())
    png_set_swap(png);
```